File: dataset/avatar/Python/Code/codeforces_404_A.py
n = int ( input ( ) )
s = [ ]
for i in range ( n ) :
    k = input ( )
    s.append ( k )
flag = 0
d1 = [ ]
d2 = [ ]
rem = set ( )
for i in range ( n ) :
    for j in range ( n ) :
        if ( i == j ) :
            d1.append ( s [ i ] [ j ] )
        if ( i == n - j - 1 ) :
            d2.append ( s [ i ] [ j ] )
        if ( i != j and i != n - j - 1 ) :
            rem.add ( s [ i ] [ j ] )
if ( len ( rem ) != 1 ) :
    print ( 'NO' )
elif ( d1 != d2 ) :
    print ( 'NO' )
elif ( len ( set ( d1 ) ) != 1 ) :
    print ( 'NO' )
elif ( set ( d1 ) == rem ) :
    print ( 'NO' )
else :
    print ( 'YES' )

File: output/deepseek_chat/avatar/algo_based_translation/python/java/successful_translation/codeforces_404_A.java

import java.util.*;

public class codeforces_404_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        scanner.nextLine();
        
        List<String> s = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            String k = scanner.nextLine();
            s.add(k);
        }
        
        int flag = 0;
        List<Character> d1 = new ArrayList<>();
        List<Character> d2 = new ArrayList<>();
        Set<Character> rem = new HashSet<>();
        
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                if (i == j) {
                    d1.add(s.get(i).charAt(j));
                }
                if (i == n - j - 1) {
                    d2.add(s.get(i).charAt(j));
                }
                if (i != j && i != n - j - 1) {
                    rem.add(s.get(i).charAt(j));
                }
            }
        }
        
        if (rem.size() != 1) {
            System.out.println("NO");
        } else if (!d1.equals(d2)) {
            System.out.println("NO");
        } else {
            Set<Character> d1Set = new HashSet<>(d1);
            if (d1Set.size() != 1) {
                System.out.println("NO");
            } else if (d1Set.equals(rem)) {
                System.out.println("NO");
            } else {
                System.out.println("YES");
            }
        }
    }
}

